project-koku/koku
# Refactor cost-explorer `order_by[date]`

## Description

* use pandas to sort the data when ordering by a specific date
* move common code into the main `ReportQueryHandler`

## Testing

1. create and ingest AWS data with 10,000 accounts
2. Restart Koku
3. Hit [http://localhost:8000/api/cost-management/v1/reports/aws/costs/?filter[limit]=1000&filter[offset]=0&group_by[account]=*&order_by[cost]=desc&order_by[date]=2022-10-02&currency=JPY&end_date=2022-10-20&start_date=2022-10-01](http://localhost:8000/api/cost-management/v1/reports/aws/costs/?filter[limit]=1000&filter[offset]=0&group_by[account]=*&order_by[cost]=desc&order_by[date]=2022-10-02&currency=JPY&end_date=2022-10-20&start_date=2022-10-01)    
4. check timings:
on main:
```
[2022-10-27 18:24:36,299] INFO None     Negative Filtering disabled acct1234567
-----      _group_by_ranks: 3.235476016998291 seconds
-----             order_by: 1.9817230701446533 seconds
```
on branch:
```
[2022-10-27 18:18:23,000] INFO None     Negative Filtering disabled acct1234567
-----      _group_by_ranks 3.3005239963531494 seconds -----
-----             order_by 1.4480640888214111 seconds -----
```
5. increase filter to 10,000:
[http://localhost:8000/api/cost-management/v1/reports/aws/costs/?filter[limit]=10000&filter[offset]=0&group_by[account]=*&order_by[cost]=desc&order_by[date]=2022-10-02&currency=JPY&end_date=2022-10-20&start_date=2022-10-01](http://localhost:8000/api/cost-management/v1/reports/aws/costs/?filter[limit]=10000&filter[offset]=0&group_by[account]=*&order_by[cost]=desc&order_by[date]=2022-10-02&currency=JPY&end_date=2022-10-20&start_date=2022-10-01)
6. check timings:
on main:
```
[2022-10-27 18:21:29,947] INFO None     Negative Filtering disabled acct1234567
-----      _group_by_ranks: 11.333613634109497 seconds
-----             order_by: 104.04372000694275 seconds
```
on branch:
```
[2022-10-27 18:19:15,323] INFO None     Negative Filtering disabled acct1234567
-----      _group_by_ranks 11.153122901916504 seconds -----
-----             order_by 9.658604145050049 seconds -----
```

## Notes

...
